File: datalayer.py
from pymongo import MongoClient
import re
import json
import sys

class DataLayer:
    def getDatabaseList(self):
        client = MongoClient('mongodb://localhost:27017/')
        return {'databases':client.database_names()}

    def getCollectionList(self,database_name):
        client = MongoClient('mongodb://localhost:27017/')
        return {'collections':client[database_name].collection_names(include_system_collections=False)}

    def getData(self, database_name, collection_name):
        client = MongoClient('mongodb://localhost:27017/')
        collection = client[database_name][collection_name]
        cursor = collection.find({}, {'_id':False})
        response = []
        cleaner = re.compile('<.*?>')
        for document in cursor:
            document['timestamp'] = document['timestamp'].isoformat()
            document['output'] = re.sub(cleaner, '', str(document['output']))
            try:
                document['input'] = re.sub(cleaner, '', str(document['input']))
            except KeyError:
                pass
            try:
                document['output'] = str(document['output']).replace("'","`")
            except KeyError:
                pass
            try:
                document['input'] = str(document['input']).replace("'","`")     
            except KeyError:
                pass
            document['intent'] = str(document['intent'])
            document['entity'] = str(document['entity'])
            response.append(document)
        return str(response).replace('"',"'").replace("'", '"').replace("`","\'")

if __name__ == '__main__':
    try:
        if len(sys.argv) > 1:
            if sys.argv[1] == 'getDatabaseList':
                print(DataLayer().getDatabaseList())        
            elif sys.argv[1] == 'getCollectionList':
                print(DataLayer().getCollectionList(sys.argv[2]))      
            elif sys.argv[1] == 'getData':
                print(DataLayer().getData(sys.argv[2], sys.argv[3]))    
    except Exception as e:
        print("Error: error while invoking",e)

File: public/datalayer.js

var deasync = require('deasync');
var MongoClient = require('mongodb').MongoClient;
var url = "mongodb://localhost:27017/";

var method = DataLayer.prototype;
function DataLayer(){
}

method.getDatabaseList = function(){  
    var response = null;
    MongoClient.connect(url, function(err, db) {
        if (err) throw err;
        db.db('test').admin().listDatabases().then(function(result){
            response = result.databases;
          }
        );
      });   
    deasync.loopWhile(function(){return response === null;});
    return JSON.stringify({"databases":response});
};

method.getCollectionList = function(databaseName){
    var response = null;
    MongoClient.connect(url, function(err, db) {
        if (err) throw err;
        var dbo = db.db(databaseName);
        dbo.listCollections().toArray(function(err, collections){
            response = collections;
      });
    });
    deasync.loopWhile(function(){return response === null;});
    return JSON.stringify({"collections":response});
};
 
method.getData = function(databaseName, collectionName){
    var response = null;
    MongoClient.connect(url, function(err, db) {
        if (err) throw err;
        var dbo = db.db(databaseName);
        dbo.collection(collectionName).find({}).toArray(function(err, result) {
          if (err) throw err;
          response = result;
          db.close();
        });
      });
    deasync.loopWhile(function(){return response === null;});
    return JSON.stringify({"data":response});
};

module.exports = DataLayer;